internal/handler/handler.go

```diff
@@ -29,13 +29,13 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	d, err := time.Parse("2006-01-02", date)
 	if err != nil {
-		writeResponse(w, http.StatusBadRequest, model.Error{"Bad request"})
+		writeResponse(w, http.StatusBadRequest, model.Error{Error: "Bad request"})
 		return
 	}
 
 	price, err := h.service.GetPrice(ticker, d)
 	if err != nil {
-		writeResponse(w, http.StatusInternalServerError, model.Error{"Internal server error"})
+		writeResponse(w, http.StatusInternalServerError, err.Error())
```

Comment:
API users don't need to observe service internal errors. We should provide something human-readable.

Reply:
Sorry, I've read the whole PR and understood the idea. Here I propose to use `errors.Is` and define messages for each error.